### PR TITLE
[AC-5495] - Allow for custom tagging when using `make release`

### PR DESCRIPTION
**Pull request by VincentHokie**
Changes Introduced in [AC-5495](https://masschallenge.atlassian.net/browse/AC-5495):

- allow use of `TAG` argument when using `make release` for custom tagging

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -403,8 +405,6 @@ release-list:
 
 
 release:
-	@git commit --allow-empty -m "generating a new release"
-	@git push
```
Comment (shanbady) [inline Makefile]: why was this removed? there was a reason for this change - without it you cannot tag repos without changes in them. for example - if you wanted to release some changes in accelerate but impact and directory should point to the latest master

**Review by shanbady — CHANGES_REQUESTED** [review]

File: create_release.sh [hunk 2]
```diff
@@ -1,7 +1,20 @@
-docker run --rm  -v"$(pwd)":/app  -ti semantic-release  -- semantic-release version
-export pwd=$(pwd)
-export TAG=$(docker run --tty=false --rm  -v$pwd:/app  -i semantic-release -- semantic-release version --noop | grep "Current version: " | cut -d ' ' -f 3 | sed -e "s/\r//")
+
+# if a tag hasn't been passed in as a parameter, create a semantic one
+if [ ! -n "$TAG" ]; then
+    git commit --allow-empty -m "generating a new release"
+	git push
```
Comment (shanbady) [inline create_release.sh]: spacing issue

**Review by shanbady — CHANGES_REQUESTED** [review]

File: create_release.sh [hunk 3]
```diff
@@ -1,7 +1,20 @@
-docker run --rm  -v"$(pwd)":/app  -ti semantic-release  -- semantic-release version
-export pwd=$(pwd)
-export TAG=$(docker run --tty=false --rm  -v$pwd:/app  -i semantic-release -- semantic-release version --noop | grep "Current version: " | cut -d ' ' -f 3 | sed -e "s/\r//")
+
+# if a tag hasn't been passed in as a parameter, create a semantic one
+if [ ! -n "$TAG" ]; then
+    git commit --allow-empty -m "generating a new release"
```
Comment (shanbady) [inline create_release.sh]: i see - this was moved here. it may be necessary regardless if we use custom tags or not

**Review by shanbady — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (VincentHokie) [reply Makefile]: put this back here, it makes sense that both with an without a tag, we should be doing these.

**Review by shanbady — APPROVED** [review]
